File: frontend/src/services/support/supportService.ts
import { fetchWithSession } from '../auth/authService';
import type { CustomerProfile, Subscription, Invoice, UsageRecord, Notification, CreditNote } from '../customer/customerService';

const API_BASE = '/support';

// ==================== TYPES ====================

export interface CustomerSearchResponse {
  customerId: number;
  fullName: string;
  email: string;
  status: 'ACTIVE' | 'INACTIVE' | 'SUSPENDED';
}

export interface CustomerDetailResponse {
  customerProfile: CustomerProfile;
  subscriptions: Subscription[];
  invoices: Invoice[];
  usageRecords: UsageRecord[];
  notifications: Notification[];
  creditNotes: CreditNote[];
}

export interface AuditLog {
  id: number;
  actor: string;
  actorRole: string;
  action: string;
  entityType: string;
  entityId: number;
  oldValue: string;
  newValue: string;
  requestId: string;
  ip: string;
  createdAt: string;
}

export interface BillingJob {
  id: number;
  jobType: string;
  status: string;
  triggeredBy: string;
  startedAt: string;
  completedAt: string;
  totalRecords: number;
  successCount: number;
  failureCount: number;
  errorSummary: string;
  createdAt: string;
}

export interface DunningRetryLog {
  id: number;
  invoiceId: number;
  paymentId: number;
  attemptNo: number;
  scheduledAt: string;
  attemptedAt: string;
  status: string;
  failureReason: string;
}

// ==================== SERVICES ====================

export const searchCustomers = async (query: string): Promise<CustomerSearchResponse[]> => {
  const response = await fetchWithSession(`${API_BASE}/customers?query=${encodeURIComponent(query)}`);
  if (!response.ok) throw new Error('Failed to search customers');
  return response.json();
};

export const getCustomerDetails = async (id: number): Promise<CustomerDetailResponse> => {
  const response = await fetchWithSession(`${API_BASE}/customers/${id}`);
  if (!response.ok) throw new Error('Failed to fetch customer details');
  return response.json();
};

export const getRecentAuditLogs = async (): Promise<AuditLog[]> => {
  const response = await fetchWithSession(`${API_BASE}/audit-logs`);
  if (!response.ok) throw new Error('Failed to fetch audit logs');
  return response.json();
};

export const getBillingJobs = async (): Promise<BillingJob[]> => {
  const response = await fetchWithSession(`${API_BASE}/billing-jobs`);
  if (!response.ok) throw new Error('Failed to fetch billing jobs');
  return response.json();
};

export const getDunningLogs = async (): Promise<DunningRetryLog[]> => {
  const response = await fetchWithSession(`${API_BASE}/dunning-logs`);
  if (!response.ok) throw new Error('Failed to fetch dunning logs');
  return response.json();
};

export const getPastDueSubscriptions = async (): Promise<Subscription[]> => {
  const response = await fetchWithSession(`${API_BASE}/subscriptions/past-due`);
  if (!response.ok) throw new Error('Failed to fetch past due subscriptions');
  return response.json();
};

File: backend/src/main/java/com/infy/billing/service/SupportService.java
package com.infy.billing.service;

import com.infy.billing.dto.support.CustomerSearchResponse;
import com.infy.billing.dto.customer.SubscriptionDTO;
import com.infy.billing.dto.support.CustomerDetailResponse;
import com.infy.billing.entity.AuditLog;
import com.infy.billing.entity.BillingJob;
import com.infy.billing.entity.DunningRetryLog;

import com.infy.billing.dto.customer.CancellationRequestDTO;
import com.infy.billing.dto.customer.CancellationResponse;
import com.infy.billing.request.ProcessCancellationRequestInput;

import java.util.List;

public interface SupportService {
    List<CustomerSearchResponse> searchCustomers(String query);

    CustomerDetailResponse getCustomerDetails(Long customerId);

    List<AuditLog> getRecentAuditLogs();

    List<BillingJob> getBillingJobs();

    List<DunningRetryLog> getDunningLogs();

    List<SubscriptionDTO> getPastDueSubscriptions();

    List<CancellationRequestDTO> getPendingCancellationRequests();

    CancellationResponse approveCancellationRequest(Long requestId, String agentEmail, ProcessCancellationRequestInput input);

    void rejectCancellationRequest(Long requestId, String agentEmail, ProcessCancellationRequestInput input);
}
